specification/dartLangSpec.tex

```diff
@@ -5837,8 +5839,8 @@ \section{Extensions}
 \item $D$ declares a type parameter named \code{E}.
 \item $D$ declares a member whose basename is the name of a type parameter
   of $D$.
-\item $D$ declares a member (static or not) with the same basename as a
-  member declared by the built-in class \code{Object}.
+\item $D$ declares a member (static or not) with the same basename as an
+  instance member declared by the built-in class \code{Object}.
```

Comment:
Let's just list them:
```latex
\begin{itemize}
  \item{} \code{==}
  \item{} \code{hashCode}
  \item{} \code{noSuchMethod}
  \item{} \code{runtimeType}
  \item{} \code{toString}
\end{itemize}
```

Maybe phrase it as:

> the same basename as one of the instance members that are implemented by all objects:
> ... the list ...

since being declared by `Object` is really descriptive rather than defining for those members (`Null` implements them too, and even `dynamic` has special-treatment for those members.)

> Or just say, "the instance members declared by the type `Object?`", to not leave `Null` out.

Reply:
We can't talk about `Object?` at this point (still waiting for reviews of null safety updates ;-), but it would indeed be nice to (1) introduce a regular name for the top type as a class (no special properties like `void` or `dynamic`), let's call it `Any`, and then talk about the members declared by `Any` where we currently talk about the members declared by `Object`.

But that's going to be handled in https://github.com/dart-lang/language/pull/2023 or some other PR which is doing part of the work done in the null safety update.

Listing the instance members directly seems to be a good choice for now, because that list would be very, very unlikely to change, and it will remain stable even if we introduce something like `Any`.

Reply:
Now using the listing.